File: ServerPhp/src/dc/DC0071.php
<?php
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0071 Controller: UI DC role permission
 */

require_once(PATH_N21LIB."/Http/Controller.php");

class DC0071 extends Controller {

  public function __construct($request, $response, $session) {
    parent::__construct($request, $response, $session);
    $this->_initFields();
  }

private function preQuery(&$params, &$where) {
    if (!empty($_REQUEST["QRY_DELETE_ALLOWED"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "DELETE_ALLOWED like :DELETE_ALLOWED";
      $params["DELETE_ALLOWED"] = $_REQUEST["QRY_DELETE_ALLOWED"];
    }
    if (!empty($_REQUEST["QRY_FETCH_ALLOWED"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "FETCH_ALLOWED like :FETCH_ALLOWED";
      $params["FETCH_ALLOWED"] = $_REQUEST["QRY_FETCH_ALLOWED"];
    }
    if (!empty($_REQUEST["QRY_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ID like :ID";
      $params["ID"] = $_REQUEST["QRY_ID"];
    }
    if (!empty($_REQUEST["QRY_INSERT_ALLOWED"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "INSERT_ALLOWED like :INSERT_ALLOWED";
      $params["INSERT_ALLOWED"] = $_REQUEST["QRY_INSERT_ALLOWED"];
    }
    if (!empty($_REQUEST["QRY_ROLE_NAME"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ROLE_NAME like :ROLE_NAME";
      $params["ROLE_NAME"] = $_REQUEST["QRY_ROLE_NAME"];
    }
    if (!empty($_REQUEST["QRY_UI_DC"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "UI_DC like :UI_DC";
      $params["UI_DC"] = $_REQUEST["QRY_UI_DC"];
    }
    if (!empty($_REQUEST["QRY_UPDATE_ALLOWED"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "UPDATE_ALLOWED like :UPDATE_ALLOWED";
      $params["UPDATE_ALLOWED"] = $_REQUEST["QRY_UPDATE_ALLOWED"];
    }
}

public function doQuery() {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),20);
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                CREATEDBY
                ,CREATEDON
                ,DELETE_ALLOWED
                ,FETCH_ALLOWED
                ,ID
                ,INSERT_ALLOWED
                ,MODIFIEDBY
                ,MODIFIEDON
                ,ROLE_NAME
                ,UI_DC
                ,UPDATE_ALLOWED
            from UI_DC_ROLE_PERMISSION  $where $orderByClause ";
    $this->logger->debug($sql);
    $rs = $this->getDbConn()->SelectLimit($sql, $limit, $start, $params);
    $rsCount = $this->getDbConn()->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    $dataOut = $this->serializeCursor($rs,array_keys($this->fields), $this->getDataFormatFetch());
    if ($this->getDataFormatFetch() == "xml" ) {header("Content-type: application/xml");}
    print $this->formatQueryResponseData($dataOut, $rsCount->fields["TOTALCOUNT"]);
} /* end function doQuery  */


public function doExport() {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),20);
    $groupBy = (!empty($_REQUEST["groupBy"]))?$_REQUEST["groupBy"]:"";
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                ID
                ,UI_DC
                ,ROLE_NAME
                ,FETCH_ALLOWED
                ,INSERT_ALLOWED
                ,UPDATE_ALLOWED
                ,DELETE_ALLOWED
                ,CREATEDON
                ,CREATEDBY
                ,MODIFIEDON
                ,MODIFIEDBY
            from UI_DC_ROLE_PERMISSION  $where $orderByClause ";
    $rs = $this->getDbConn()->Execute($sql, $params);
    $rsCount = $this->getDbConn()->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    if (!empty($_REQUEST["_p_disp_cols"])) {
      $columns = explode("|",$_REQUEST["_p_disp_cols"]);
    } else {
      $columns = array_keys($this->fields);
    }
    if ($this->getDataFormatExport() == "csv" ) {
      $dataOut = $this->serializeCursor($rs,$columns,"csv");
    } else {
      $dataOut = $this->serializeCursor($rs,$columns,"xml");
      $dataOut = "<records>".$dataOut."</records>";
      $dataOut = "<queryParams>".$this->serializeMap($params,"xml")."</queryParams>".$dataOut;
      $dataOut = "<columnDef>".$this->columnDefForExport($columns,$this->fields,true).$this->columnDefForExport(array_diff(array_keys($params), $columns),$this->fields,false)."</columnDef>".$dataOut;
      $dataOut = "<staticText>".$this->exportLocalizedStaticText()."</staticText>".$dataOut;
      $dataOut = "<groupBy>".$groupBy."</groupBy>".$dataOut;
      $dataOut = "<reportData  title=\"".$this->getDcTitle()."\" by=\"".$_SESSION["user"]["userName"]."\" on=\"".date(DATE_FORMAT)."\">".$dataOut."</reportData>";
    }
    $this->beginExport();
    print $dataOut;
    $this->endExport();
} /* end function doExport  */

public function fetchRecord() {
    $this->record["ID"] = $this->getRequestParam("ID");
    if (empty($this->record["ID"])) { throw new Exception("Missing value for primary key field ID in DC0071.fetchRecord().");}
    $pkArray = array("ID" => $this->record["ID"]);
    $this->findByPk($pkArray, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function fetchRecord */

public function doInsert() {
    $this->populateRecordFromRequest(); 
    $this->record["_p_record_status"] = $this->getRequestParam("_p_record_status");
    $this->record["_p_store_recId"] = $this->getRequestParam("_p_store_recId");
    $sql = "insert into UI_DC_ROLE_PERMISSION(
                 DELETE_ALLOWED
                ,FETCH_ALLOWED
                ,ID
                ,INSERT_ALLOWED
                ,ROLE_NAME
                ,UI_DC
                ,UPDATE_ALLOWED
            ) values ( 
                 :DELETE_ALLOWED
                ,:FETCH_ALLOWED
                ,:ID
                ,:INSERT_ALLOWED
                ,:ROLE_NAME
                ,:UI_DC
                ,:UPDATE_ALLOWED
    )";
    $stmt = $this->getDbConn()->prepare($sql);
    $_seq = $this->getDbConn()->execute("select SEQ_UIDCROLEPRMS_ID.nextval seq_val from dual")->fetchRow();
    $this->record["ID"] = $_seq["SEQ_VAL"];
    $this->getDbConn()->Execute($stmt, $this->record);
    $pkArray = array("ID" => $this->record["ID"]);
    $this->findByPk($pkArray, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function doInsert */

public function doUpdate() {
    $this->populateRecordFromRequest();
    $this->record["_p_record_status"] = $this->getRequestParam("_p_record_status");
    $this->record["_p_store_recId"] = $this->getRequestParam("_p_store_recId");
    $sql = "update UI_DC_ROLE_PERMISSION set 
                 DELETE_ALLOWED=:DELETE_ALLOWED
                ,FETCH_ALLOWED=:FETCH_ALLOWED
                ,ID=:ID
                ,INSERT_ALLOWED=:INSERT_ALLOWED
                ,UPDATE_ALLOWED=:UPDATE_ALLOWED
    where 
           ID= :ID
    ";
    $stmt = $this->getDbConn()->prepare($sql);
    $this->getDbConn()->Execute($stmt, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function doUpdate */

public function doDelete() {
    $this->record["ID"] = $this->getRequestParam("ID");
    if (empty($this->record["ID"])) { throw new Exception("Missing value for primary key field ID in DC0071.doDelete().");}
    $sql = "delete from UI_DC_ROLE_PERMISSION where 
           ID= :ID
    ";
    $stmt = $this->getDbConn()->prepare($sql);
    $this->getDbConn()->Execute($stmt, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function doDelete */

public function initNewRecord() {
    $this->populateRecordFromRequest();
    $this->record["_p_record_status"] = "insert";
    $this->setFieldInitialValues($this->record, "DC0071");
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function initNewRecord */

private function findByPk(&$pkCols, &$record) {
    $sql = "select 
                CREATEDBY
                ,CREATEDON
                ,DELETE_ALLOWED
                ,FETCH_ALLOWED
                ,ID
                ,INSERT_ALLOWED
                ,MODIFIEDBY
                ,MODIFIEDON
                ,ROLE_NAME
                ,UI_DC
                ,UPDATE_ALLOWED
            from UI_DC_ROLE_PERMISSION 
         where 
           ID= :ID
            ";
    $rs = $this->getDbConn()->Execute($sql, $pkCols);
    $row = $rs->FetchRow();
    CollectionUtils::copyMapValues($record, $row);
} /* end function findByPk  */


public function callProcedure($pName) {
  $this->logger->debug("Start: ".$this->dcName.".callProcedure(".$pName.")");
    $this->populateRecordFromRequest();
    print "{success:true, data:".json_encode($this->record)."}";
    $this->logger->debug("End: ".$this->dcName.".callProcedure");
} /* end function callProcedure */


private function _initFields() {
  $this->fields = array(
  "CREATEDBY" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"CREATEDON" => array(parent::FLDPROP_DATA_TYPE => "DATE")
  ,"DELETE_ALLOWED" => array(parent::FLDPROP_DATA_TYPE => "BOOLEAN")
  ,"FETCH_ALLOWED" => array(parent::FLDPROP_DATA_TYPE => "BOOLEAN")
  ,"ID" => array(parent::FLDPROP_DATA_TYPE => "NUMBER")
  ,"INSERT_ALLOWED" => array(parent::FLDPROP_DATA_TYPE => "BOOLEAN")
  ,"MODIFIEDBY" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"MODIFIEDON" => array(parent::FLDPROP_DATA_TYPE => "DATE")
  ,"ROLE_NAME" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"UI_DC" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"UPDATE_ALLOWED" => array(parent::FLDPROP_DATA_TYPE => "BOOLEAN")
);
}

}
?>

File: trunk/ServerJava/src/net/nan21/ebs/dc/DC0071.java
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0071 DC Controller: UI DC role permission
 */

package net.nan21.ebs.dc;


import java.sql.Connection;
import java.sql.ResultSet;
import java.util.Date;
import java.util.HashMap;
import java.util.Properties;
import javax.servlet.http.HttpServletResponse;
import net.nan21.ebs.lib.CollectionUtils;
import net.nan21.ebs.lib.AbstractDataControl;
import net.nan21.ebs.lib.FieldDef;
import net.nan21.ebs.lib.HttpRequest;
import net.nan21.ebs.lib.HttpSession;
import net.nan21.ebs.lib.IDataControl;
import net.nan21.ebs.lib.DbManager;

public class DC0071 extends AbstractDataControl implements IDataControl {

  public void init (HttpRequest request, HttpServletResponse response, HttpSession session, DbManager dbm) throws Exception {
    this._initFields();
    super.init(request, response, session, dbm);
  }

private void preQuery() {
    if (this.request.getParam("QRY_DELETE_ALLOWED") != null && !this.request.getParam("QRY_DELETE_ALLOWED").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("DELETE_ALLOWED like :DELETE_ALLOWED");
      this.queryParams.put("DELETE_ALLOWED",(String)this.request.getParam("QRY_DELETE_ALLOWED"));
    }
    if (this.request.getParam("QRY_FETCH_ALLOWED") != null && !this.request.getParam("QRY_FETCH_ALLOWED").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("FETCH_ALLOWED like :FETCH_ALLOWED");
      this.queryParams.put("FETCH_ALLOWED",(String)this.request.getParam("QRY_FETCH_ALLOWED"));
    }
    if (this.request.getParam("QRY_ID") != null && !this.request.getParam("QRY_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ID like :ID");
      this.queryParams.put("ID",(String)this.request.getParam("QRY_ID"));
    }
    if (this.request.getParam("QRY_INSERT_ALLOWED") != null && !this.request.getParam("QRY_INSERT_ALLOWED").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("INSERT_ALLOWED like :INSERT_ALLOWED");
      this.queryParams.put("INSERT_ALLOWED",(String)this.request.getParam("QRY_INSERT_ALLOWED"));
    }
    if (this.request.getParam("QRY_ROLE_NAME") != null && !this.request.getParam("QRY_ROLE_NAME").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ROLE_NAME like :ROLE_NAME");
      this.queryParams.put("ROLE_NAME",(String)this.request.getParam("QRY_ROLE_NAME"));
    }
    if (this.request.getParam("QRY_UI_DC") != null && !this.request.getParam("QRY_UI_DC").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("UI_DC like :UI_DC");
      this.queryParams.put("UI_DC",(String)this.request.getParam("QRY_UI_DC"));
    }
    if (this.request.getParam("QRY_UPDATE_ALLOWED") != null && !this.request.getParam("QRY_UPDATE_ALLOWED").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("UPDATE_ALLOWED like :UPDATE_ALLOWED");
      this.queryParams.put("UPDATE_ALLOWED",(String)this.request.getParam("QRY_UPDATE_ALLOWED"));
    }
}

public void doQuery() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " CREATEDBY"+
               " ,to_char(CREATEDON,'"+this.DATE_FORMAT_DB+"') CREATEDON"+
               " ,DELETE_ALLOWED"+
               " ,FETCH_ALLOWED"+
               " ,ID"+
               " ,INSERT_ALLOWED"+
               " ,MODIFIEDBY"+
               " ,to_char(MODIFIEDON,'"+this.DATE_FORMAT_DB+"') MODIFIEDON"+
               " ,ROLE_NAME"+
               " ,UI_DC"+
               " ,UPDATE_ALLOWED"+
           " from UI_DC_ROLE_PERMISSION  "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoQuery(sql);
} 


public void doExport() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " ID"+
               " ,UI_DC"+
               " ,ROLE_NAME"+
               " ,FETCH_ALLOWED"+
               " ,INSERT_ALLOWED"+
               " ,UPDATE_ALLOWED"+
               " ,DELETE_ALLOWED"+
               " ,to_char(CREATEDON,'"+this.DATE_FORMAT_DB+"') CREATEDON"+
               " ,CREATEDBY"+
               " ,to_char(MODIFIEDON,'"+this.DATE_FORMAT_DB+"') MODIFIEDON"+
               " ,MODIFIEDBY"+
           " from UI_DC_ROLE_PERMISSION  "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoExport(sql);
}

 	public void fetchRecord()  throws Exception  {
     this.populateRecordPkFromRequest();
     this.findByPk();
       this.writeResultFetchRecord();	 
   }

public void doInsert()  throws Exception {
  this.populateRecordFromRequest(); 
  this.populateRecordWithClientSpecific();
    String sql = "insert into UI_DC_ROLE_PERMISSION("+
               "  DELETE_ALLOWED"+
               " ,FETCH_ALLOWED"+
               " ,ID"+
               " ,INSERT_ALLOWED"+
               " ,ROLE_NAME"+
               " ,UI_DC"+
               " ,UPDATE_ALLOWED"+
           " ) values ( "+
               "  :DELETE_ALLOWED"+
               " ,:FETCH_ALLOWED"+
               " ,:ID"+
               " ,:INSERT_ALLOWED"+
               " ,:ROLE_NAME"+
               " ,:UI_DC"+
               " ,:UPDATE_ALLOWED"+
    ")";
    this.record.put("ID",   dbm.getSequenceNextValue("SEQ_UIDCROLEPRMS_ID")  );
    dbm.executeStatement(sql, this.record);
    this.populateRecordPkFromRecord();
    this.findByPk();
    this.writeResultDoInsert();
}

public void doUpdate()  throws Exception {
    this.populateRecordFromRequest();
    this.populateRecordWithClientSpecific();
    String sql = "update UI_DC_ROLE_PERMISSION set "+
               "  DELETE_ALLOWED=:DELETE_ALLOWED"+ 
               " ,FETCH_ALLOWED=:FETCH_ALLOWED"+ 
               " ,ID=:ID"+ 
               " ,INSERT_ALLOWED=:INSERT_ALLOWED"+ 
               " ,UPDATE_ALLOWED=:UPDATE_ALLOWED"+ 
   " where "+ 
     "      ID= :ID"+
    "";
    dbm.executeStatement(sql, this.record);
    this.populateRecordPkFromRecord();
    this.findByPk();
    this.writeResultDoUpdate();	
}

public void doDelete() throws Exception {
    this.populateRecordPkFromRequest();
   String sql = "delete from UI_DC_ROLE_PERMISSION where "+
     "      ID= :ID"+
    "";
    dbm.executeStatement(sql, this.recordPk);
   this.writeResultDoDelete();
}

public void initNewRecord() throws Exception {
    this.populateRecordFromRequest();
    this.record.put("_p_record_status", "insert");
    this.writeResultInitNewRecord();
}

private void findByPk()  throws Exception {
    String sql = "select "+ 
               " CREATEDBY"+
               " ,to_char(CREATEDON,'"+this.DATE_FORMAT_DB+"') CREATEDON"+
               " ,DELETE_ALLOWED"+
               " ,FETCH_ALLOWED"+
               " ,ID"+
               " ,INSERT_ALLOWED"+
               " ,MODIFIEDBY"+
               " ,to_char(MODIFIEDON,'"+this.DATE_FORMAT_DB+"') MODIFIEDON"+
               " ,ROLE_NAME"+
               " ,UI_DC"+
               " ,UPDATE_ALLOWED"+
           " from UI_DC_ROLE_PERMISSION "+
        " where "+
     "      ID= :ID"+ 
          "";
    this.record = dbm.executeQuery(sql, this.recordPk).get(0); 
} 


public void callProcedure(String pName)  throws Exception {
    this.populateRecordFromRequest();
}


	private void  _initFields() {
	  this.fields = new HashMap<String, FieldDef>();
	  this.fields.put("CREATEDBY", new FieldDef("STRING"));
	  this.fields.put("CREATEDON", new FieldDef("DATE"));
	  this.fields.put("DELETE_ALLOWED", new FieldDef("BOOLEAN"));
	  this.fields.put("FETCH_ALLOWED", new FieldDef("BOOLEAN"));
	  this.fields.put("ID", new FieldDef("NUMBER"));
	  this.fields.put("INSERT_ALLOWED", new FieldDef("BOOLEAN"));
	  this.fields.put("MODIFIEDBY", new FieldDef("STRING"));
	  this.fields.put("MODIFIEDON", new FieldDef("DATE"));
	  this.fields.put("ROLE_NAME", new FieldDef("STRING"));
	  this.fields.put("UI_DC", new FieldDef("STRING"));
	  this.fields.put("UPDATE_ALLOWED", new FieldDef("BOOLEAN"));
	  String[] _pkFields = {"ID"};
	  this.pkFields = _pkFields;
	}

public void doCustomAction(String action) {}
}
